src/development/LocalTest/Services/LocalApp/Implementation/LocalAppFile.cs

```diff
@@ -0,0 +1,87 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+using Altinn.Platform.Storage.Interface.Models;
+
+using LocalTest.Configuration;
+using LocalTest.Services.LocalApp.Interface;
+using LocalTest.Services.Localtest.Interface;
+
+namespace LocalTest.Services.LocalApp.Implementation
+{
+    public class LocalAppFile : ILocalApp
+    {
+        private readonly LocalPlatformSettings _localPlatformSettings;
+        public LocalAppFile(IOptions<LocalPlatformSettings> localPlatformSettings)
+        {
+            _localPlatformSettings = localPlatformSettings.Value;
+        }
+
+        public async Task<string?> GetXACMLPolicy(string appId)
+        {
+            return await File.ReadAllTextAsync(Path.Join(GetAppPath(appId), $"App/config/authorization/policy.xml"));
+        }
+
+        public async Task<Application?> GetApplicationMetadata(string appId)
+        {
+            var filename = Path.Join(GetAppPath(appId), $"App/config/applicationmetadata.json");
+            if (File.Exists(filename))
+            {
+                var content = await File.ReadAllTextAsync(filename, Encoding.UTF8);
+                return JsonConvert.DeserializeObject<Application>(content);
+            }
+            
+            return null;
+        }
+
+        public async Task<Dictionary<string, Application>> GetApplications()
+        {
+            var ret = new Dictionary<string, Application>();
+            // Get list of apps from the configured folder AppRepositoryBasePath
+            string path = _localPlatformSettings.AppRepositoryBasePath;
+
+            if (!Directory.Exists(path))
+            {
+                // No directory found.
+                return ret;
```

Comment:
Here we could return an empty dictionary to make it more obvious what we return
```suggestion
                return new Dictionary<string, Application>();
```
The declaration of `ret` can then be moved down below the if.